lua/wire/server/wirelib.lua

```diff
@@ -527,6 +581,13 @@ function WireLib.TriggerOutput(ent, oname, value, iter)
 	if (not ent.Outputs) then return end
 
 	local output = ent.Outputs[oname]
+
+	local ty = WireLib.DT[output.Type]
+	if ty and not ty.Validator(value) then
+		-- Not copying here is fine since data types are immutable outside E2.
+		value = ty.Zero
```

Comment:
Should return a copy of 'Zero'. Maybe change 'ty.Zero' into a function that returns a new zero so this line becomes `value = ty.Zero()`

Reply:
I'm not sure how ports handle mutable types, so maybe I'm wrong.

Reply:
Wire just doesn't mutate any data types, and E2 `table.Copy`s the types before using them and outputting them.

But since this is a PR to fix addon creators potentially messing stuff up, might as well do this too in case they mutate the default value.